### ISSUE TITLE
Atlas/Query node naming/types mismatch

### ISSUE THREAD

**Issue by kdembler** [Joystream/joystream#1698]
During the initial integration of Atlas and the query node, some incompatibilities were found. This issue compiles the ones I've found so far. Please mind that it's possible there are some that I've missed and will be found only later once it's actually possible to test Atlas with the real query node.

### Connection queries naming

Atlas uses a pluralized form of a noun with a `Connection` suffix, the query node uses a singular form of that same noun. E.g. Atlas - `videosConnection`, query node - `videoConnection`.

__Recommendation__: Update the query node to use the plural form.
This isn't clearly defined by the Relay connection spec (it constantly uses just the plural noun, without the suffix), but the GraphQL docs use the form used by Atlas (https://graphql.org/learn/pagination/#complete-connection-model)

### Mismatch on the field name

List of mismatches with recommendation on which version to settle on:
- `Video.publishedOnJoystreamAt` _(Atlas)_ <-> `Video.createdAt` _(query node)_ - use __query node__ version
- `Video.thumbnailURL` _(Atlas)_ <-> `Video.thumbnailUrl` _(query node)_ - use __query node__ version
- `Channel.coverPhotoURL` _(Atlas)_ <-> `Channel.coverPhotoUrl` _(query node)_ - use __query node__ version
- `Channel.avatarPhotoURL` _(Atlas)_ <-> `Channel.avatarPhotoUrl` _(query node)_ - use __query node__ version
- `Channel.handle` _(Atlas)_ <-> `Channel.title` _(query node)_ - use __Atlas__ version

### Mismatch on the field type

- `Channel.coverPhotoUrl`, `Channel.avatarPhotoUrl`:
In the schema used by Atlas, both of these are nullable and that was our assumption during implementation (e.g. provide default channel avatar/cover). The query node marks both of those as non-nullable.
__Recommendation__: Make it consistent with the constraints of the content directory - if the fields aren't required, make them nullable

- `Video.channel`:
Atlas assumes the channel will never be null for a video, i.e. every video is always published by _some_ channel. The query node allows the channel to be nullable - that would have a lot of UI implications in Atlas.
__Recommendation__: Make the channel field non-nullable

- `Video.category` - Nullable in the query node, required in Atlas.
__Recommendation:__ Changing this doesn't have any implication for Atlas at this point, so I think it's mostly a product question @bedeho 

- `Video.publishedBeforeJoystream` - `String` in Atlas, `Int` in the query node.
__Recommendation__: Use the `DateTime` (same as the `createdAt`) for consistency. Also possibly we should rename the field to `publishedBeforeJoystreamAt` so it's clear it contains a timestamp and is not a boolean flag for example.

There's also a couple of nullable fields in the query node that in my understanding should definitely be non-nullable and rather are this way because of some error:
- `Video.media`
- `Video.license`
- `VideoMedia.location`
- `VideoMedia.encoding`

### Mismatch on type assumptions

- `MediaLocation`
Atlas assumes this type to be a simple union - `JoystreamVideoMediaLocation | HTTPVideoMediaLocation` - it has to be one of these. However, the query node has a separate `MediaLocation` type with fields `httpMediaLocation` and `joystreamMediaLocation`. In general, this structure is fine, but in the query node, both of those fields are nullable. This means that the API gives no guarantee that at least one of these values will be present. This poses a challenge to a client as it can theoretically get no location at all.
__Reccomendation:__ Use some kind of union that will guarantee a value is present at all times

- `License`
Basically the same situation as above, but with `KnownLicense` and `UserDefinedLicense`.
__Recommendation:__ Same as above, but with a lower priority
